# python:3.7-slim - May 31, 2019
FROM python@sha256:e191d73f7f87dfff6f723655ff5222e1c90c9e67bc727a5c5a4d4cd45943efd4

RUN apt-get update && \
        apt-get install -y \
        bash \
        build-essential \
        curl \
        gcc \
        git \
        libjpeg-dev \
        libffi-dev \
        libpq-dev \
        libtiff-dev \
        libssl-dev \
        libz-dev \
        musl-dev \
        netcat-traditional \
        paxctl \
        python3-dev

COPY docker/django-start.sh /usr/local/bin
RUN  chmod +x /usr/local/bin/django-start.sh

# docker-compose must pass in the host UID here so that the volume
# permissions are correct
ARG USERID
RUN adduser --disabled-password --gecos "" --uid "${USERID?USERID must be supplied}" gcorn

RUN paxctl -cm /usr/local/bin/python
COPY securethenews/dev-requirements.txt /requirements.txt
RUN pip install --require-hashes -r /requirements.txt

RUN  mkdir /deploy && \
    chown -R gcorn: /deploy

EXPOSE 8000
USER gcorn

CMD ["/usr/local/bin/django-start.sh"]
